4/bin/entrypoint.sh

```diff
@@ -1,7 +1,20 @@
 #!/bin/bash
 
-set -e
-set -o pipefail
+set -euo pipefail
+
+declare -a args
+
+add_env_var_as_env_prop() {
+  if [ ! -z "$1" ]; then
+    args+=("-D$2=$1")
+  fi
+}
+
+add_env_var_as_env_prop "${SONAR_HOST_URL:-}" "sonar.host.url"
+add_env_var_as_env_prop "${SONAR_LOGIN:-}" "sonar.login"
+add_env_var_as_env_prop "${SONAR_PASSWORD:-}" "sonar.password"
+add_env_var_as_env_prop "${SONAR_USER_HOME:-}" "sonar.userHome"
 
 export SONAR_USER_HOME="$PWD/.sonar"
```

Comment:
Shouldn't we put this export into Dockerfile? There is no way that user could override this value before you added passing properties to scanner through envs

Reply:
> There is no way that user could override this value before you added passing properties to scanner through envs

I don't understand what you mean by that

Nevertheless, your comment made me thinking and I came to the conclusions below.

Setting the `SONAR_USER_HOME` to this location by default is important to run the scanner properly: the project dir is the only place we can safely assume the scanner will have permissions to write to for sure.

Removing this from the script will stop it from working standalone. It will become dependent on the DockerFile and I believe this is a maintenance pitfall as things become more fragile (and "dependencies are bad").

I don't see this as a constraint of the image as a whole: if the user overrides the entry point to run `/bin/bash` or run the scanner binary directly, I see no reason for this var to be set in either cases (first case: doesn't matter, second case: I'm going full manual, the vanilla setup I should get).

Similarly, if user overwrites the entry point or extends the image and still uses our script, I think it ought to do the right thing by default.